conda/repodata.py

```diff
@@ -0,0 +1,99 @@
+# (c) Continuum Analytics, Inc. / http://continuum.io
+# All Rights Reserved
+#
+# conda is distributed under the terms of the BSD 3-clause license.
+# Consult LICENSE.txt or http://opensource.org/licenses/BSD-3-Clause.
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+from .core import repodata as _impl
+from .connection import CondaSession
+from os.path import join
+
+
+class RepoData(object):
+    """This object represents all the package metainfo of a single channel."""
+
+    _data = {}
+
+    @staticmethod
+    def enable(url, name, priority, cache_dir=None):
+        RepoData._data[url] = RepoData(url, name, priority, cache_dir)
+
```

Comment:
Oh, I didn't comment here.  I don't understand the word "enable" here.  And why expose `enable` and `get`?  Why not make `get` a lazy call to what you have as `enable`? Or even better, why not just have the `RepoData` instantiation take care of all of that for you?